libraries/Microsoft.Bot.Builder.Dialogs/DialogContainer.cs

```diff
@@ -49,6 +50,31 @@ public virtual Dialog FindDialog(string dialogId)
             return this.Dialogs.Find(dialogId);
         }
 
+        /// <summary>
+        /// Called when an event has been raised, using `DialogContext.emitEvent()`, by either the current dialog or a dialog that the current dialog started.
+        /// </summary>
+        /// <param name="dc">The dialog context for the current turn of conversation.</param>
+        /// <param name="e">The event being raised.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>True if the event is handled by the current dialog and bubbling should stop.</returns>
+        public override async Task<bool> OnDialogEventAsync(DialogContext dc, DialogEvent e, CancellationToken cancellationToken)
+        {
+            var handled = await base.OnDialogEventAsync(dc, e, cancellationToken).ConfigureAwait(false);
+
+            // Trace unhandled "versionChanged" and "error" events.
+            if (!handled && (e.Name == DialogEvents.VersionChanged || e.Name == DialogEvents.Error))
+            {
+                var traceMessage = $"Unhandled dialog event: {e.Name}. Active Dialog: {dc.ActiveDialog.Id}";
```

Comment:
For unhandled `DialogEvents.Error` events I don't _think_ we need to send an additional trace, as I would expect that the Dialogs throw re-throw the Exception. (The behavior shown in [DialogManager](https://github.com/microsoft/botbuilder-dotnet/blob/614697f884b57301403cf01b522a605524794a45/libraries/Microsoft.Bot.Builder.Dialogs/DialogManager.cs#L231-L241))

That said... We don't appear to natively emit an error DialogEvent from inside the SDK except from DialogManager. This implies that if an error event is emitted via another case, the developer has specifically emitted the error and should be handling it themselves.

> ### _e.g._
>When a DialogContainer subclass's OnDialogEventAsync() is called with a DialogEvent (DialogEvents.Error) and DialogManager **isn't** being used, that DialogEvent should come from the customer's code.
>
> The customer may be using DialogExtensions.RunAsync() with a Dialog subclass that utilizes DialogEvents.

Since both DialogManager and _non-DialogManager_ cases should be accounted for by either the SDK or the developer (rethrow vs business logic), I think the TelemetryTrace and TraceActivity for Errors is unnecessary.

___

Can you verify that coverage exists for `Dialog.OnDialogEventAsync()` and then layer more tests for `DialogContainer.OnDialogEventAsync()`?

Coveralls indicates [coverage](https://coveralls.io/builds/32145424/source?filename=/libraries/Microsoft.Bot.Builder.Dialogs/Dialog.cs#L200) for `Dialog.OnDialogEventAsync()`, but Coveralls has been spotty for us in the past, so I'm taking the report with a large grain of salt 😄

Reply:
Remove the check / trace for error events and left just versionChanged